src/prdoc.rs

```diff
@@ -196,9 +196,31 @@ fn compare_deps(
         .and_then(|d| d.as_table())
         .unwrap_or(&t);
 
-    for (name, _) in deps {
-        let (_old_pkg, old_dep, old_root_dep) = get_dep(workspace, name, old, old_root)?;
-        let (new_pkg, new_dep, new_root_dep) = get_dep(workspace, name, new, new_root)?;
+    for name in deps.iter().map(|(name, _)| name).collect::<HashSet<_>>() {
+        let ((_old_pkg, old_dep, old_root_dep), (new_pkg, new_dep, new_root_dep)) = match (
+            get_dep(workspace, name, old, old_root),
+            get_dep(workspace, name, new, new_root),
+        ) {
+            (Err(_), Ok(_)) | (Ok(_), Err(_)) => {
+                // removed or added
+                changes.push(DepChange {
+                    name: c.name().to_string(),
+                    path: c
+                        .root()
+                        .strip_prefix(workspace.root())
+                        .unwrap()
+                        .to_path_buf(),
+                    dep: name.to_string(),
+                    breaking: false,
+                });
+                continue;
+            }
+            (Err(_), Err(_)) => {
+                // strange, not found in both?
```

Comment:
May be an issue with dep renames not properly handled? E.g. `foo = {version = "1", package = "bar"}`. I'll need to look into it.

Reply:
And what exactly is "dep renames not properly handled"? How should it be done correctly?

Ok, I change it here to throw error with exact message: https://github.com/paritytech/parity-publish/pull/43/commits/8eb07d52b7aecad8e1dc8faae28c6122d59a87ba

So now, it won't fail in most cases when adding or removing dependencies, but it will fail in this specific corner case.

Reply:
aha, ok :), so continue/skip (I can put just some log there) or fail here?

Reply:
(Originally posted on the wrong account)

> Sorry for the confusion, I was just commenting on the comment. Dependency renames are when the dependency has a package = and this exists under different names to the registry and local dependency tree. I'm just noting it as a possible issue that I'll look into in the future. It's not a user error,

> aha, ok :), so continue/skip (I can put just some log there) or fail here?

I think skipping is fine for now if it fixes the CI. I'll look into it further when I get the chance.

Reply:
And no log as the message implies there's user action to take when there's not.